From eb7ff5c08c5de843c6a3dae80e95c57dba7a505a Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Fri, 10 Mar 2023 18:37:23 +0000
Subject: [PATCH] i40e-v2.22.18: don't install auxiliary module on
 modules_install_no_aux

modules_install_no_aux inconveniently installs the auxiliary module
despite what the name would suggest.

This happens because the auxiliary module is saved to the same location
as the normal driver module, so when the driver Makefile calls
modules_install with the Linux kernel Makefile, both modules are installed.

The Linux kernel modules_install reads the .mod files generated
during the module build process to determine the modules to install.
These .mod files are simply text files with the path to the .ko file.
(and also a .o file).

Avoid installing the auxiliary module during modules_install_no_aux
by removing the .mod file that tells modules_install that it exists.

---
 src/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Makefile b/src/Makefile
index ac4e0ca..109959a 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -125,6 +125,7 @@ modules_install: default
 # files outside of the build root. Thus, it must not update initramfs, or run depmod.
 modules_install_no_aux:
 	@echo "Installing modules..."
+	@rm -f .tmp_versions/intel_auxiliary.mod
 	@+$(call kernelbuild,modules_install)

 # After installing all the files, perform necessary work to ensure the system
--
2.35.6

